base/pm/pm.go

```
@@ -504,3 +505,28 @@ func (pm *PM) resultCallback(cmd *core.Command, result *core.JobResult) {
 		handler(cmd, result)
 	}
 }
+
+//System is a wrapper around core.system
+func (pm *PM) System(bin string, args ...string) (*core.JobResult, error) {
+	runner, err := pm.RunCmd(&core.Command{
+		ID:      uuid.New(),
+		Command: process.CommandSystem,
+		Arguments: core.MustArguments(
+			process.SystemCommandArguments{
+				Name: bin,
+				Args: args,
+			},
+		),
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	job := runner.Wait()
+	if job.State != core.StateSuccess {
+		return nil, fmt.Errorf("exited with error (%s): %v", job.State, job.Streams)
```

Why the need to return the error if it can already be got from the job? also this will make it harder for the caller to get any of the streams if he wants to handle the error.